service_capacity_modeling/models/org/netflix/elasticsearch.py

```diff
@@ -0,0 +1,471 @@
+import logging
+import math
+from decimal import Decimal
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+
+from service_capacity_modeling.interface import AccessConsistency
+from service_capacity_modeling.interface import AccessPattern
+from service_capacity_modeling.interface import CapacityDesires
+from service_capacity_modeling.interface import CapacityPlan
+from service_capacity_modeling.interface import CapacityRequirement
+from service_capacity_modeling.interface import certain_float
+from service_capacity_modeling.interface import certain_int
+from service_capacity_modeling.interface import Clusters
+from service_capacity_modeling.interface import DataShape
+from service_capacity_modeling.interface import Drive
+from service_capacity_modeling.interface import FixedInterval
+from service_capacity_modeling.interface import Instance
+from service_capacity_modeling.interface import Interval
+from service_capacity_modeling.interface import QueryPattern
+from service_capacity_modeling.interface import RegionContext
+from service_capacity_modeling.interface import Requirements
+from service_capacity_modeling.models import CapacityModel
+from service_capacity_modeling.models.common import compute_stateful_zone
+from service_capacity_modeling.models.common import simple_network_mbps
+from service_capacity_modeling.models.common import sqrt_staffed_cores
+from service_capacity_modeling.models.common import working_set_from_drive_and_slo
+from service_capacity_modeling.stats import dist_for_interval
+
+
+logger = logging.getLogger(__name__)
+
+
+def _target_rf(desires: CapacityDesires, user_copies: Optional[int]) -> int:
+    if user_copies is not None:
+        assert user_copies > 1
+        return user_copies
+
+    # Due to the relaxed durability and consistency requirements we can
+    # run with RF=2
+    consistency = desires.query_pattern.access_consistency.same_region
+    if (
+        desires.data_shape.durability_slo_order.mid < 1000
+        and consistency is not None
+        and consistency.target_consistency != AccessConsistency.read_your_writes
+    ):
+        return 2
+    return 3
+
+
+# Looks like Elasticsearch (Lucene) uses a tiered merge strategy of 10
+# segments of 512 megs per
+# https://lucene.apache.org/core/8_1_0/core/org/apache/lucene/index/TieredMergePolicy.html#setSegmentsPerTier(double)
```

Comment:
ES specific defaults are set here: https://github.com/elastic/elasticsearch/blob/v6.8.2/server/src/main/java/org/elasticsearch/index/MergePolicyConfig.java
It is a tiered merge strategy with 10 segments per tier and max merged segment size is 5GB.

Reply:
:+1: I think this IO estimate should be accurate then (although if there are no cloud drives involved it shouldn't really matter).

Reply:
Agree